client/src/Components/HOCs/AuthenticationRestrictedRoute.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import { connect } from 'react-redux'
+import { Route, Redirect } from 'react-router';
+
+export const AuthenticationRestrictedRoute = ({
+  component: Component,
+  render: RenderedComponent, // Renaming render prop to avoid confusion + better readability.
+  redirectTo,
+
+  UserReducer, // Not passed to Route, but used to evaluate whether the user is logged in.
```

Comment:
Maybe we could call this `currentUser` or something similar, not sure. 🤔

Reply:
Great idea, would make for greater readability. Changing now.